### PR TITLE
test: Fix flaky HostSuite.TestAddFailingJob

### PR DESCRIPTION
**Pull request by titanous**
We don't guarantee the order that events are delivered in.

### CODE REVIEW

File: test/test_host.go [hunk 1]
```diff
@@ -63,26 +63,27 @@ func (s *HostSuite) TestAddFailingJob(t *c.C) {
...
 			}
 		case <-time.After(30 * time.Second):
 			t.Fatal("timed out waiting for job event")
 		}
 	}
-	t.Assert(actual, c.HasLen, 2)
-	t.Assert(actual[0].Event, c.Equals, host.JobEventCreate)
-	t.Assert(actual[1].Event, c.Equals, host.JobEventError)
-	jobErr := actual[1].Job.Error
+	t.Assert(actual[host.JobEventCreate], c.Not(c.IsNil))
```
Comment (lmars) [inline test/test_host.go]: Should this check that `actual[host.JobEventError]` is not nil, given that is what we use for further assertions below?

Comment (titanous) [inline test/test_host.go]: Done.

File: test/test_host.go [hunk 2]
```diff
@@ -63,28 +63,31 @@ func (s *HostSuite) TestAddFailingJob(t *c.C) {
...
-	t.Assert(actual, c.HasLen, 2)
-	t.Assert(actual[0].Event, c.Equals, host.JobEventCreate)
-	t.Assert(actual[1].Event, c.Equals, host.JobEventError)
-	jobErr := actual[1].Job.Error
-	t.Assert(jobErr, c.NotNil)
-	t.Assert(*jobErr, c.Equals, `host: invalid job partition "nonexistent"`)
+	t.Assert(actual[host.JobEventCreate], c.NotNil)
+	e := actual[host.JobEventError]
+	c.Assert(e, c.NotNil)
+	c.Assert(e.Job, c.NotNil)
```
Comment (lmars) [inline test/test_host.go]: These need to be `t.Assert`

**Comment by lmars** [comment]
LGTM pending comment

File: test/test_host.go (hunk 2 above)
Comment (titanous) [inline test/test_host.go]: Fixed, thanks.

**Pull request merged**